src/Microsoft.Framework.CommandLineUtils.Sources/CommandLine/CommandLineApplication.cs

```diff
@@ -332,6 +332,10 @@ public void ShowHelp(string commandName = null)
             else
             {
                 target = Commands.SingleOrDefault(cmd => string.Equals(cmd.Name, commandName, StringComparison.OrdinalIgnoreCase));
+                if (target == null)
+                {
+                    throw new ArgumentException(string.Format("Error: unexpected value '{0}' for option 'help'", commandName), "commandName");
+                }
```

Comment:
If we run into this case, it means that the user is trying to do something like
```
dnu help invalid_subcommand
```
We may want to make the message more friendly:
```
$"'{commandName}' is not a subcommand of '{cmd.Parent.Name}'"
```